app/recipe/serializers.py

```diff
@@ -0,0 +1,49 @@
+from rest_framework import serializers
+
+from core.models import Ingredient, Recipe
+
+
+class IngredientSerializer(serializers.ModelSerializer):
+    """Serializers for ingredient objects"""
+
+    class Meta:
+        model = Ingredient
+        fields = ('name',)
+
+
+class RecipeSerializer(serializers.ModelSerializer):
+    """Serializers for recipe objects"""
+    ingredients = IngredientSerializer(many=True)
+
+    class Meta:
+        model = Recipe
+        fields = ('id', 'name', 'description', 'ingredients')
+        read_only_fields = ('id',)
+
+    def create(self, validated_data):
+        ingredients = validated_data.pop('ingredients')
+        recipe = Recipe.objects.create(**validated_data)
+        for ing in ingredients:
+            Ingredient.objects.get_or_create(recipe=recipe, **ing)
+        recipe.save()
+        recipe.refresh_from_db()
```

Comment:
Why do you call `save` and `refresh_from_db`?

Doesn't the `create` call in line 25 store the object in the database? 
https://docs.djangoproject.com/en/3.1/ref/models/querysets/#create

Reply:
Yes, you are right I actually don't need them. Thank you for spotting!